docs/ops/generation/EyeLike_9.md

```diff
@@ -0,0 +1,125 @@
+## EyeLike <a name="EyeLike"></a> {#openvino_docs_ops_generation_EyeLike_9}
+
+**Versioned name**: *EyeLike-9*
+
+**Category**: *Generation*
+
+**Short description**: *EyeLike* operation generates identity matrix.
+
+**Detailed description**:
+
+*EyeLike* operation generates matrix with ones on the diagonal and zeros everywhere else. Index of the diagonal to be populated with ones is given by `diagonal_index`: `output[i, i + diagonal_index] = 1`.
+
+
+Example 1. *EyeLike* output with `output_type` = f32:
+
+``` 
+num_rows = 3
+
+output  = [[1. 0. 0.]
+           [0. 1. 0.]
+           [0. 0. 1.]]
+```
+
+Example 2. *EyeLike* output with `output_type` = i32, `diagonal_index` = 2:
+
+``` 
+num_rows = 3
+
+num_columns = 4
+
+output  = [[0 0 1 0]
+           [0 0 0 1]
+           [0 0 0 0]]
+```
+
+Example 3. *EyeLike* output with `output_type` = i32, `diagonal_index` = -1:
+
+``` 
+num_rows = 3
+
+num_columns = 4
+
+output  = [[0 0 0 0]
+           [1 0 0 0]
+           [0 1 0 0]]
+```
+
+Example 4. *EyeLike* output with `output_type` = f16, `diagonal_index` = 5:
+
+``` 
+num_rows = 2
+
+output  = [[0. 0.]
+           [0. 0.]]
+```
+
+**Attributes**:
+
+* *output_type*
+
+    * **Description**: the type of the output
+    * **Range of values**: any numeric type
+    * **Type**: string
+    * **Required**: *Yes*
+
+* *diagonal_index*
+
+    * **Description**: index of the diagonal to be populated. A positive value refers to an upper diagonal and a negative value refers to a lower diagonal. Value `0` populates the main diagonal.
+    * **Range of values**: any integer value
+    * **Type**: int
+    * **Default value**: *0*
+    * **Required**: *No*
+
+
+**Inputs**:
+
+*   **1**: `num_rows` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. **Required.**
+
+*   **2**: `num_columns` - scalar or 1D tensor with 1 non-negative element of type *T_NUM* describing the number of rows in matrix. Optionally, with default value equals to `num_rows`.
+
+
+**Outputs**:
+
+* **1**: A tensor with type specified by the attribute *output_type*. The shape is `[num_rows, num_columns]`
+
+**Types**
+
+* *T_NUM*: `int32` or `int64`.
+
+**Examples**
+
+*Example 1*
+
+```xml
+<layer ... name="EyeLike" type="EyeLike">
+    <data output_type="i32"/>
+    <input>
+        <port id="0" precision="I32"/>  <!-- num rows -->
+    </input>
+    <output>
+        <port id="3" precision="I32" names="EyeLike:0">
+            <dim>-1</dim>
```

Comment:
I think it is better to add example with static output shape. Also, is this operation really can have dynamic output shape?

Reply:
Just imagine that inputs are constants we can write static output shape, I agree. Done

As for really dynamic outputs. 
If inputs (`row_number` and `column_number`) are parameters, output matrix shape will be dynamic.
For example, TF has `row_number` and `column_number` as parameters. So in TF this node can has dynamic output shape.
Also in ONNX this op has input - 2D matrix that can be dynamic. So we can say that ONNX support dynamic  `row_number` and `column_number` (because it's size of dynamic input shape).

Reply:
Ok, thanks for explanation! I think it is important moment and we should mention it in out specification and have related example.

Can one dimension be dynamic and one - static?

Reply:
Added new examples with dynamic output, thanks!